Add OSXChromiumDriver and support extra browser arguments in run-benchmark

emw-apple opened the pull request:
#### fa4a5245bfbbbbd12c6fd9fea70b4efc83f2023a
<pre>
Add OSXChromiumDriver and support extra browser arguments in run-benchmark
<a href="https://bugs.webkit.org/show_bug.cgi?id=253553">https://bugs.webkit.org/show_bug.cgi?id=253553</a>
rdar://106389711

Reviewed by Stephanie Lewis.

Support running benchmarks against Chromium in run-benchmark. Chromium
is like Chrome, except that it doesn&apos;t use the
--enable-field-trial-config switch, because field trials are enabled by
default.

Additionally, allow extra arguments to be passed to the browser process
through run-benchmark, to facilitate customized testing.

* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_chrome_driver.py:
(OSXChromiumDriver): Added.

* Tools/Scripts/webkitpy/benchmark_runner/run_benchmark.py:
(config_argument_parser): Add benchmark_args argument.

Remaining changes are all plumbing benchmark_args through to the browser
driver.

* Tools/Scripts/webkitpy/benchmark_runner/benchmark_runner.py:
(BenchmarkRunner.__init__):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver.__init__):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver_factory.py:
(BrowserDriverFactory.create):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py:
(OSXBrowserDriver._launch_process):
(run_benchmark_plan):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner.__init__):
* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_unittest.py:
(BrowserPerfDashRunnerTest.test_can_construct_runner_object_minimum_parameters):
(FakeBrowserDriver.__init__):

Canonical link: <a href="https://commits.webkit.org/261697@main">https://commits.webkit.org/261697@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c55a199dc6ec4568d44003e821ab5108d5c373b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/112229 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/21364 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/858 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4028 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120871 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/116292 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22712 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/12677 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/4792 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117992 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16874 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100056 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/105290 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115655 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98831 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/597 "4 failures") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45881 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13768 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/639 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/94053 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/14447 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/10128 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19811 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52641 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8157 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/16242 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard reviewed (commented):
Looks like `test-webkitpy` is unhappy, per EWS results.

emw-apple committed:
Add OSXChromiumDriver and support extra browser arguments in run-benchmark
https://bugs.webkit.org/show_bug.cgi?id=253553
rdar://106389711

Reviewed by Stephanie Lewis.

Support running benchmarks against Chromium in run-benchmark. Chromium
is like Chrome, except that it doesn't use the
--enable-field-trial-config switch, because field trials are enabled by
default.

Additionally, allow extra arguments to be passed to the browser process
through run-benchmark, to facilitate customized testing.

* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_chrome_driver.py:
(OSXChromiumDriver): Added.

* Tools/Scripts/webkitpy/benchmark_runner/run_benchmark.py:
(config_argument_parser): Add benchmark_args argument.

Remaining changes are all plumbing benchmark_args through to the browser
driver.

* Tools/Scripts/webkitpy/benchmark_runner/benchmark_runner.py:
(BenchmarkRunner.__init__):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver.__init__):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver_factory.py:
(BrowserDriverFactory.create):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_browser_driver.py:
(OSXBrowserDriver._launch_process):
(run_benchmark_plan):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner.__init__):
* Tools/Scripts/webkitpy/browserperfdash/browserperfdash_unittest.py:
(BrowserPerfDashRunnerTest.test_can_construct_runner_object_minimum_parameters):
(FakeBrowserDriver.__init__):

Canonical link: https://commits.webkit.org/261697@main
